include/pru/am572x.h

```diff
@@ -0,0 +1,748 @@
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef __PRU_AM572X_H__
+#define __PRU_AM572X_H__
+
+
+/* Local memory map (TRM section 30.1.4.4.1.2) */
+#define __LOCAL_DATA_RAM_BASE		0x00000000
+#define __PEER_DATA_RAM_BASE		0x00002000
+#define __SHARED_DATA_RAM_BASE		0x00010000
+#define __INTC_BASE			0x00020000
+#define __PRU0_CTRL_REGS_BASE		0x00022000
+#define __PRU1_CTRL_REGS_BASE		0x00024000
+#define __CFG_BASE			0x00026000
+#define __UART0_BASE			0x00028000
+#define __IEP_BASE			0x0002E000
+#define __ECAP0_BASE			0x00030000
+#define __MII_RT_CFG_BASE		0x00032000
+#define __MII_MDIO_BASE			0x00032400
+#define __EXTERNAL_PRU_SS_BASE  0x00040000
+#define __SYS_OCP_HP0_BASE		0x00080000
+
+/* ====================== CTABLE base addresses ====================== */
+#pragma ctable_entry 0 0x00020000
+#pragma ctable_entry 1 0x48040000
+#pragma ctable_entry 2 0x4802a000
+#pragma ctable_entry 3 0x00030000
+#pragma ctable_entry 4 0x00026000
+#pragma ctable_entry 5 0x48060000
+#pragma ctable_entry 6 0x48030000
+#pragma ctable_entry 7 0x00028000
+#pragma ctable_entry 8 0x46000000
+#pragma ctable_entry 9 0x4a100000
+#pragma ctable_entry 10 0x48318000
+#pragma ctable_entry 11 0x48022000
+#pragma ctable_entry 12 0x48024000
+#pragma ctable_entry 13 0x48318000
+#pragma ctable_entry 14 0x481cc000
+#pragma ctable_entry 15 0x481d0000
+#pragma ctable_entry 16 0x481a0000
+#pragma ctable_entry 17 0x4819c000
+#pragma ctable_entry 18 0x48300000
+#pragma ctable_entry 19 0x48302000
+#pragma ctable_entry 20 0x48304000
+#pragma ctable_entry 21 0x00032400
+#pragma ctable_entry 22 0x480c8000
+#pragma ctable_entry 23 0x480ca000
+/* 24-31 are user-defined base addresses. */
+
+/* ====================== PRU CTRL module ====================== */
+#define PRU_CTRL_CONTROL_OFFS		0x00
+#define PRU_CTRL_STATUS_OFFS		0x04
+#define PRU_CTRL_WAKEUP_EN_OFFS		0x08
+#define PRU_CTRL_CYCLE_OFFS		0x0c
+#define PRU_CTRL_STALL_OFFS		0x10
+#define PRU_CTRL_CTBIR0_OFFS		0x20
+#define PRU_CTRL_CTBIR1_OFFS		0x24
+#define PRU_CTRL_CPPR0_OFFS		0x28
+#define PRU_CTRL_CPPR1_OFFS		0x2c
+
+struct __PRU_CONTROL_S {
+	const volatile unsigned int SOFT_RST_N		: 1;
+	volatile unsigned int ENABLE			: 1;
+	volatile unsigned int SLEEPING			: 1;
+	volatile unsigned int COUNTER_ENABLE		: 1;
+	const volatile unsigned int __reserved_1	: 4;
+	volatile unsigned int SINGLE_STEP		: 1;
+	const volatile unsigned int __reserved_2	: 5;
+	const volatile unsigned int BIG_ENDIAN		: 1;
```

Comment:
newlib defines BIG_ENDIAN as a macro in ${PREFIX}/pru/include/machine/endian.h .When I tried to build an rpmsg example with your header I got an error:

```
/home/dinux/bin/opt/pru-gcc/pru/include/pru/am572x.h:123:30: error: expected identifier or '(' before numeric constant
  123 |  const volatile unsigned int BIG_ENDIAN  : 1;
      |                              ^~~~~~~~~~
```

Try to come up with a solution which does not change newlib. Example: rename this bit field to PRU_BIG_ENDIAN

Reply:
Fixed in 0949dfe24a7525df5eb78e57c1b826e73fb61634